rust/agama-cli/src/questions.rs

```diff
@@ -71,11 +72,14 @@ async fn ask_question() -> Result<(), ServiceError> {
     let question = serde_json::from_reader(std::io::stdin())?;
 
     let created_question = client.create_question(&question).await?;
-    let answer = client.get_answer(created_question.generic.id).await?;
+    let Some(id) = created_question.generic.id else {
+        return Err(ServiceError::QuestionNotExist(0));
```

Comment:
What does that '0' mean? It looks like the error is not exactly that the question does not exist.

Reply:
To be honest this should not happen as it is assigned in that create_question from created question path. I am not sure if we should have something like Programatic error or something that for cases which should not happen and user cannot do naything  with it